src/game.cpp

```diff
@@ -9826,15 +9826,26 @@ void game::on_move_effects()
 {
     // TODO: Move this to a character method
     if( !u.is_mounted() ) {
+        const item muscle( "muscle" );
         if( u.lifetime_stats.squares_walked % 8 == 0 ) {
             if( u.has_active_bionic( bionic_id( "bio_torsionratchet" ) ) ) {
                 u.charge_power( 1 );
             }
+            for( const bionic_id &bid : u.get_bionic_fueled_with( muscle ) ) {
+                if( u.has_active_bionic( bid ) ) {
+                    u.charge_power( muscle.fuel_energy() * ( bid->fuel_efficiency / 100 ) );
+                }
+            }
         }
         if( u.lifetime_stats.squares_walked % 160 == 0 ) {
             if( u.has_bionic( bionic_id( "bio_torsionratchet" ) ) ) {
                 u.charge_power( 1 );
             }
+            for( const bionic_id &bid : u.get_bionic_fueled_with( muscle ) ) {
+                if( u.has_active_bionic( bid ) ) {
+                    u.charge_power( muscle.fuel_energy() * ( bid->fuel_efficiency / 100 ) );
```

Comment:
The term `charge_power( ... * efficiency / 100 )` seems to be repeated again and again and should be in a function.

Reply:
Changed fuel_efficiency to a float